quick-start/setup.js

```diff
@@ -0,0 +1,282 @@
+   const loginSpinner = logger.spinner( CURRENT_STEP );
+   loginSpinner.start();
+
+   try {
+      await exec( `echo ${ DOCKER_TOKEN } | docker login -u cs --password-stdin https://${ DOCKER_ENDPOINT }` );
+   } catch( err ) {
+      loginSpinner.stop();
+      
+      if ( err.stderr.includes( "403 Forbidden" ) ) {
+         throw new SetupError( error.wrongToken );
+      }
+      throw new Error( err.stderr );
+   }
+
+   loginSpinner.stop();
+   logger.stepInfo( CURRENT_STEP );
+}
+
+async function pullDockerImage() {
+   CURRENT_STEP = step.dockerPull;
+   const downloadSpinner = logger.spinner( CURRENT_STEP );
+   downloadSpinner.start();
+
+   try {
+      await exec( `docker pull ${ DOCKER_ENDPOINT }/cs:latest` );
+   } catch( err ) {
+      downloadSpinner.stop();
+      throw new Error( err.stderr );
+   }
+
+   downloadSpinner.stop();
+   logger.stepInfo( CURRENT_STEP );
+}
+
+function editDockerComposeFile() {
+   const yaml = require( 'js-yaml' );
+   
+   CURRENT_STEP = step.editFile;
+   
+   try {
+      let dockerComposeFile = fs.readFileSync( './docker-compose.yml', 'utf8' );
+      let dockerComposeObject = yaml.load( dockerComposeFile );
+
+      dockerComposeObject.services[ 'ckeditor-cs' ].environment.LICENSE_KEY = LICENSE_KEY;
+      dockerComposeObject.services[ 'ckeditor-cs' ].environment.ENVIRONMENTS_MANAGEMENT_SECRET_KEY = ENV_SECRET;
+      dockerComposeObject.services[ 'ckeditor-cs' ].ports[ 0 ] = `${ CS_PORT }:8000`;
+      dockerComposeObject.services[ 'node-server' ].ports[ 0 ] = `${ NODE_PORT }:3000`;
+       
+      dockerComposeFile = yaml.dump( dockerComposeObject );
+      fs.writeFileSync( './docker-compose.yml', dockerComposeFile, 'utf8' );
+   } catch ( err ) {
+      throw new Error( err );
+   }
+   logger.stepInfo( CURRENT_STEP );
+}
+
+async function startDockerContainers() {
+   CURRENT_STEP = step.dockerUp;
+   CLEANUP_NEEDED  = true;
+   const dockerSpinner = logger.spinner( CURRENT_STEP );
+   dockerSpinner.start();
+
+   const spawn = require( 'child_process' ).spawn;
+   const dockerImages = spawn( "docker-compose", [ "up", "--build" ] );
+   
+   dockerImages.output = '';
+   dockerImages.stdout.on( 'data', ( data ) => {
+      dockerImages.output += data.toString();
+   } );
+
+   return new Promise( ( resolve, reject ) => {
+     
+      const serversAvailabilityCheck = setInterval( () => {
+         if ( dockerImages.output.includes( 'Server is listening on port 8000.' ) &&
+         dockerImages.output.includes( 'Node-server is listening on port 3000' ) ) {
+            clearInterval( serversAvailabilityCheck );
+            clearTimeout( timeout );
+            dockerSpinner.stop();
+            logger.stepInfo( CURRENT_STEP );
+            resolve();
+         }
+         
+         if ( dockerImages.output.includes( 'Wrong license key.' ) ) {
+            dockerSpinner.stop();
+            reject( new SetupError( error.wrongLicense ) );
+         }
+      }, 100 ); 
```

Comment:
:thinking:  Ooo interesting approach but IMO better will be to use `/health` endpoints

On-premises has this endpoint and you could add very simple endpoint to Node server.  

![](https://user-images.githubusercontent.com/1795926/116688100-7ced4a80-a9b6-11eb-83e0-cdc22d4624c6.png)

```
app.use( '/health' , ( req, res ) => res.send( { uptime: process.uptime(), timestamp: Date.now() } ) );
```

or something similar